lib/find-vs2017.js

```diff
@@ -0,0 +1,47 @@
+const log = require('npmlog')
+  , execSync = require('child_process').execSync
+  , path = require('path')
+
+var hasCache = false
+  , cache = null
+
+function findVS2017() {
+  if (hasCache)
+    return cache
+
+  hasCache = true
+
+  const ps = 'PowerShell -ExecutionPolicy Unrestricted -Command '
+  const csFile = path.join(__dirname, 'Find-VS2017.cs')
+  const psQuery = ps + '"&{Add-Type -Path \'' + csFile + '\'; [VisualStudioConfiguration.Main]::Query()}" 2>&1'
+
+  var vsSetup
+  try {
+    const vsSetupRaw = execSync(psQuery, { encoding: 'utf8' })
```

Comment:
`var` for consistency, also given that node-gyp still works for very old versions of Node